packages/flutter/lib/src/material/animated_icons/animated_icons.dart

```diff
@@ -92,26 +90,26 @@ class AnimatedIcon extends StatelessWidget {
   ///
   ///  * [SemanticsProperties.label], which is set to [semanticLabel] in the
   ///    underlying [Semantics] widget.
-  final String semanticLabel;
+  final String? semanticLabel;
 
   /// The text direction to use for rendering the icon.
   ///
   /// If this is null, the ambient [Directionality] is used instead.
   ///
   /// If the text direction is [TextDirection.rtl], the icon will be mirrored
   /// horizontally (e.g back arrow will point right).
-  final TextDirection textDirection;
+  final TextDirection? textDirection;
 
   static final _UiPathFactory _pathFactory = () => ui.Path();
 
   @override
   Widget build(BuildContext context) {
     final _AnimatedIconData iconData = icon as _AnimatedIconData;
     final IconThemeData iconTheme = IconTheme.of(context);
-    final double iconSize = size ?? iconTheme.size;
-    final TextDirection textDirection = this.textDirection ?? Directionality.of(context);
-    final double iconOpacity = iconTheme.opacity;
-    Color iconColor = color ?? iconTheme.color;
+    final double iconSize = size ?? iconTheme.size!;
```

Comment:
These `!` on the iconTheme property are unfortunate. We know that `IconTheme.of` only returns theme data where that is true, I wish we could express that in code better. 

Maybe add an `assert(iconTheme.isConcrete)` above this line to make it clear to the reader?

Reply:
assert added